tests/v2/validation/schemas/schemas_test.go

```diff
@@ -0,0 +1,469 @@
+//go:build (validation || infra.any || cluster.any || extended) && !sanity && !stress
+
+package schemas
+
+import (
+	"context"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/kubeapi/customresourcedefinitions"
+	"github.com/rancher/shepherd/extensions/kubectl"
+	"github.com/rancher/shepherd/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+type SchemaChangesTestSuite struct {
+	suite.Suite
+	client  *rancher.Client
+	session *session.Session
+	cluster *management.Cluster
+}
+
+func (sc *SchemaChangesTestSuite) TearDownSuite() {
+	sc.session.Cleanup()
+}
+
+func (sc *SchemaChangesTestSuite) SetupSuite() {
+	sc.session = session.NewSession()
+
+	client, err := rancher.NewClient("", sc.session)
+	require.NoError(sc.T(), err)
+
+	sc.client = client
+
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(sc.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(sc.client, clusterName)
+	require.NoError(sc.T(), err, "Error getting cluster ID")
+	sc.cluster, err = sc.client.Management.Cluster.ByID(clusterID)
+	require.NoError(sc.T(), err)
+}
+
+func (sc *SchemaChangesTestSuite) TestSchemaEndpointLocalCluster() {
+	subSession := sc.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Access the schemas endpoint on the local cluster and verify that it can be accessed successfully without any errors.")
+	schemasCollection, err := sc.client.Steve.SteveType(schemaSteveType).List(nil)
+	require.NoError(sc.T(), err)
+	log.Infof("Number of schemas: %d", len(schemasCollection.Data))
+
+	log.Info("Verify that each listed schema's resourceFields are set to null and ensure the schemas endpoint exclusively lists top-level types without including child schemas for resources.")
+	for _, schema := range schemasCollection.Data {
+		schemaID := schema.JSONResp["id"].(string)
+		if !exceptionMap[schemaID] {
+			require.Nil(sc.T(), schema.JSONResp["resourceFields"], "ResourceFields should be null for schema "+schemaID)
+
+			if match, _ := regexp.MatchString(`\.[vV][0-9]+`, schemaID); match {
+				require.Fail(sc.T(), "Child schema should not be listed")
+			}
```

Comment:
This part is a little confusing to me, if its not from the exception map and is nil, we check if child schema is listed or not is it?

Reply:
No, if the schema does not exist in the exception map, we then verify whether the resourceFields is null and subsequently check if child schema is listed or not

Reply:
Discussed this offline, so resolving this comment.